src/components/ContactUsForm/index.jsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * ContactUsForm consists of validation and elements of contact form, imports Netlifyform for post data.
+ * The base code that is used in both netlifyForm and index are implemented from https://github.com/jon-fm/react-ssg-netlify-forms-demo
+ */
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import NetlifyForm from "../NetlifyForm";
+import { Col, Row } from "antd";
+import { RiSendPlaneFill } from "react-icons/ri";
+import "./index.css";
+
+const ContactUsForm = () => {
+  const [t] = useTranslation();
+  const [submissionMessage, setSubmissionMessage] = useState("");
+  // beforeSubmit is for validations and disabling submit button
+  const [processing, setProcessing] = useState(false);
+  const beforeSubmit = async () => {
+    if (
+      contactFormValues.name.length > 0 &&
+      contactFormValues.email.length > 0
+    ) {
+      setProcessing(true);
+      // Wait 2 seconds to simulate async delay
+      await new Promise((resolve) => setTimeout(resolve, 2000));
+      return true;
+    } else {
+      return false;
+    }
+  };
+  // TODO: get submission message to translate files.
+  // afterSubmit is for submission message and enable submit button
+  const afterSubmit = () => {
+    setSubmissionMessage("Thanks for messaging us!");
+    setProcessing(false);
+  };
+  // handles value changes of contact form
+  const handleChange = (e) =>
+    setContactFormValues({
+      ...contactFormValues,
+      [e.target.name]: e.target.value,
+    });
+  const [contactFormValues, setContactFormValues] = useState({
+    name: "",
+    email: "",
+    message: "",
+  });
+  return (
+    <div className="contactUsForm">
+      <NetlifyForm
+        formName="contact"
+        contactFormValues={contactFormValues}
+        beforeSubmit={beforeSubmit}
+        afterSubmit={afterSubmit}
+        automaticHoneypot={true}
+      >
+        <span className="contactUsForm__result">{submissionMessage}</span>
```

Comment:
Thanks to Tarkan for taking care of most of the comments, I have a small one here, 

Do you think we need to consider using conditional rendering of this so it doesn't render an empty span when submittionMessage is empty?

Reply:
Thanks for your review Yaman, I used span, because it's a block element. But I'll try to add conditional rendering to the span as you've mentioned.